### ISSUE TITLE
Make error: [_install] Error 255 (Cask is unreliable)

### ISSUE THREAD

**Issue by doomchild** [hlissner/.emacs.d#8]
Dependency realgud failed to install: Wrong type argument: stringp, (\, "https://melpa.org/packages/")
Output:
Setting `package-selected-packages' temporarily since "emacs -q" would overwrite customizations

make: **\* [_install] Error 255

**Comment by hlissner** [comment]
Unfortunately, cask (what I use to manage packages) has proven to be very unreliable. When this happens (and it happens often for me), I've found the only way to fix it is to reboot (maybe just logging out and logging back in might work). I can't say why, only that I suspect some corrupted state is preserved between `cask install` calls that screws with things. I haven't been able to find where that state is kept. I'm only certainly it's not in `/tmp`.

TL;DR, could you try to reboot before running `make install` again? Just to rule out the possibility that this is Cask's doing.

**Comment by doomchild** [comment]
I had uncommented the package lines at the top (since I'm using Emacs 25 now) before the failures, so I commented them back up, and it stopped happening.  Super weird.

**Comment by hlissner** [comment]
Eyup, that's another issue I've inconsistently encountered as well and, again, I suspect Cask is to blame. I'm close to simply rolling my own plugin management system, but until that levvy breaks, I'm afraid we're stuck with it.

I'll keep this issue open until I've found a satisfactory solution to this.

**Comment by doomchild** [comment]
Is there a reason you don't want to use the built-in package system?

**Comment by hlissner** [comment]
The main reason: some plugins I use aren't available on any ELPA source, or are github-only forks. package.el can't pull straight from repos, nor can it manage them. And they fluctuate so much (and can sometimes be so numerous), I think it would be impractical to use submodules to keep track of them.

But I also prefer to manage plugins outside of Emacs, so that I can automate the process with tools like `make`. (And as a side-note, it means faster startup times. Woo!)

**Comment by doomchild** [comment]
Okay, that makes sense.  Despite having used emacs for a while now, I'm really only just now starting to expand the number of packages I use, so I'm not surprised that I haven't run into the same problem.

**Comment by hlissner** [comment]
v2.0 is done, and now has it's own package manager (powered by package.el and QUELPA), which has turned out _much_ faster and more reliable than cask, so I'll close this issue now.
